x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/SlimResults.java

```diff
@@ -90,11 +90,11 @@ public Object predictedValue() {
 
     @Override
     void doXContentBody(XContentBuilder builder, Params params) throws IOException {
-        builder.startArray(resultsField);
+        builder.startObject(resultsField);
```

Comment:
we should make this an object with `token: weight`. Same for the inference results. This fits better with rank_features